packages/babel-parser/src/tokenizer/index.js

```diff
@@ -168,6 +169,18 @@ export default class Tokenizer extends LocationParser {
     return curr;
   }
 
+  nextTokenStart(): number {
+    const thisTokEnd = this.state.pos;
+    skipWhiteSpace.lastIndex = thisTokEnd;
+    const skip = skipWhiteSpace.exec(this.input);
+    // $FlowIgnore: The skipWhiteSpace ensures to match any string
+    return thisTokEnd + skip[0].length;
+  }
+
```

Comment:
@JLHwung In most cases this will not improve the performance. Using regex for this purpose may have opposite effect. I looked at the lexer code and ... well... I understand why you do this.., but you should only need to use current index like this `this.input.charCodeAt(this.index)`.  Refactor the lexer into a `while loop` and you get everything like this for free without any need for regular expressions. You only need to do `index + 1` if you hit `0x20`. If you need the length, store the length before and after the iteration in a locale variable. Doing it like this will also get rid of some overhead like `[0]` and `.length`.  For this case you simply use a locale variable at the start of the function 
`let length = 0` and inside the loop you do this `length++`. And your return will be `TokEnd + length`.

Reply:
You shouldn't blindly trust jsperf, but a regex can be super fast if done in a single operation / task, but in the case of Babel parser this have multiple purposes and slow down. In the long run a `while loop` is the best solution when it comes to a parser. Even if you super optimize the regular expression.
I just give you a friendly tip. Everyone code differently :) But to illustrate this run a benchmark against Acorn. You see Babel parser is 1x slower (*estimated*). Reason I guess is that even the small things haven't been optimized when adding new stuff, and changing the small things as you do in this case will have larger impact then you think. In a positive way.
You can also run a benchmark against my parser too - Meriyah. I guess Babel parser is 4 -5x slower (*estimated*).